fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/client/event/lifecycle/v1/ClientTaskScheduler.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.event.lifecycle.v1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.minecraft.client.MinecraftClient;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Allows the scheduling of events after a specified number of client ticks.
+ */
+@Environment(EnvType.CLIENT)
+public class ClientTaskScheduler {
+	static {
+		// Don't add this the the event handler until the class is loaded
+		ClientTickEvents.START_CLIENT_TICK.register(ClientTaskScheduler::runExecutables);
+	}
+
+	private static final List<Pair<Integer, List<ClientTickEvents.StartTick>>> executables = new ArrayList<>();
```

Comment:
Yeah, you should use int to object hash map (either java, fastutil int map, or guava multimap) instead. 

Also I would add a field in this class indicating the current tick number and increment that field than decrementing all the integer keys of the list. Using that field is okay and it doesn't hurt from integer overflow either.